hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestCreateTable.scala

```diff
@@ -272,4 +278,48 @@ class TestCreateTable extends TestHoodieSqlBase {
       )
     }
   }
+
+  test("Test Create Table From Exist Hoodie Table") {
+    withTempDir{tmp =>
+      // Write a table by spark dataframe.
+      val tableName = generateTableName
+      import spark.implicits._
+      val df = Seq((1, "a1", 10, 1000, "2021-08-02")).toDF("id", "name", "value", "ts", "dt")
+      df.write.format("hudi")
+        .option(HoodieWriteConfig.TABLE_NAME.key, tableName)
+        .option(TABLE_TYPE_OPT_KEY.key, COW_TABLE_TYPE_OPT_VAL)
+        .option(RECORDKEY_FIELD_OPT_KEY.key, "id")
+        .option(PRECOMBINE_FIELD_OPT_KEY.key, "ts")
+        .option(PARTITIONPATH_FIELD_OPT_KEY.key, "dt")
+        .option(KEYGENERATOR_CLASS_OPT_KEY.key, classOf[ComplexKeyGenerator].getName)
+        .option(HoodieWriteConfig.INSERT_PARALLELISM.key, "1")
+        .option(HoodieWriteConfig.UPSERT_PARALLELISM.key, "1")
+        .mode(SaveMode.Overwrite)
+        .save(tmp.getCanonicalPath)
+
+      // Create a table over the exist old table.
+      spark.sql(
+        s"""
+           |create table $tableName using hudi
+           | options (
+           | primaryKey = 'id',
+           | preCombineField = 'ts'
+           |)
+           |partitioned by (dt)
+           |location '${tmp.getCanonicalPath}'
+           |""".stripMargin)
+      checkAnswer(s"select id, name, value, ts, dt from $tableName")(
+        Seq(1, "a1", 10, 1000, "2021-08-02")
+      )
+      // Check the missing properties for spark sql
+      val metaClient = HoodieTableMetaClient.builder()
+        .setBasePath(tmp.getCanonicalPath)
+        .setConf(spark.sessionState.newHadoopConf())
+        .build()
+      val properties = metaClient.getTableConfig.getProps.asScala.toMap
+      assertResult(true)(properties.contains(HoodieTableConfig.HOODIE_TABLE_CREATE_SCHEMA.key))
+      assertResult("dt")(properties(HoodieTableConfig.HOODIE_TABLE_PARTITION_FIELDS_PROP.key))
+      assertResult("ts")(properties(HoodieTableConfig.HOODIE_TABLE_PRECOMBINE_FIELD_PROP.key))
```

Comment:
Can we do some basic table operations as well. 
1. create hudi table w/ spark data source with few records(lets call this as old record). 
2. create the same table in spark-sql. 
   a. ensure table props match
   b. INSERT into works 
   c. UPDATE -> updates both old records and new records 
   d. DELETE -> again, ensure both old records and new records that matched are deleted.. 
   e. MERGE INTO -> ensure there is a match for both old record and a new record as well. both should get updated.

Reply:
Yes, will add more test case for this.